lib/ansible/executor/task_executor.py

```diff
@@ -134,7 +134,10 @@ def _clean_res(res):
                 if isinstance(res, UnsafeProxy):
                     return res._obj
                 elif isinstance(res, binary_type):
-                    return to_text(res, errors='surrogate_or_strict')
+                    try:
+                        return to_text(res, errors='surrogate_or_strict')
+                    except:
+                        return bytes(res)
```

Comment:
@Yannig @jimi-c also thinks that  doing this generally will impact matching on the results (not just the diff output results).  He suggests catching the unicodeerror and raising a nicer error about only being able to use diff with utf8 file contents.

If we're able to tell that we're using diff here, though, maybe we could run to_text([..], error='surrogate_or_replace') just for the diff entry?  @jimi-c?